velox/docs/functions/spark/array.rst

```diff
@@ -27,6 +27,30 @@ Array Functions
 
         SELECT array_intersect(array(1, 2, 3), array(1, 3, 5)); -- [1,3]
 
+.. function:: array_max(array(E)) -> E
+
+    Returns maximum non-NULL element of the array. Returns NULL if array is empty or all elements are NULL.
+    When E is DOUBLE or REAL, returns NaN if any element is NaN. ::
+
+        SELECT array_max(ARRAY [1, 2, 3]); -- 3
+        SELECT array_max(ARRAY [-1, -2, -2]); -- -1
+        SELECT array_max(ARRAY [-1, -2, NULL]); -- -1
+        SELECT array_max(ARRAY []); -- NULL
+        SELECT array_max(ARRAY [-0.0001, -0.0002, -0.0003, float('nan')]); -- NaN
+
+.. function:: array_min(array(E)) -> E
+
+    Returns minimum non-NULL element of the array. Returns NULL if array is empty or all elements are NULL.
+    When E is DOUBLE or REAL, NaN value is considered greater than any non-NaN value. ::
```

Comment:
>Returns NULL if array is empty or contains a NULL element.
> NaN value is considered greater than any non-NaN value.

This doesn't seem to hold true completely.
NaN seems to always be returned in both min_by and max_by even if NULL exists which also means its simultaneously considered both highest and lowest value.
Tried these on presto java:
```
SELECT array_max(ARRAY [nan(), 1.0]); => NaN 
SELECT array_max(ARRAY [NULL, nan(), 1.0]); => NaN 

SELECT array_min(ARRAY [nan(), 1.0]); => NaN 
SELECT array_min(ARRAY [NULL, nan(), 1.0]); => NaN 
```